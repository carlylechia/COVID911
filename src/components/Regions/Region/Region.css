.states-container {
  display: flex;
  justify-content: space-between;
}

.states-container:nth-child(even) {
  background: rgb(222, 49, 99);
}

.states-container:nth-child(odd) {
  background: #d70040;
}
