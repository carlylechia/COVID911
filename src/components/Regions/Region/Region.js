import PropTypes from 'prop-types';
import './Region.css';

const Region = ({ name, newCases }) => (
  <div className="states-container">
    <p className="">{name}</p>
    <p className="">{newCases}</p>
  </div>
);
Region.propTypes = {
  name: PropTypes.string.isRequired,
  newCases: PropTypes.number.isRequired,
};
export default Region;
