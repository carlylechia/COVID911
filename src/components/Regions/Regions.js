import { useSelector, useDispatch, shallowEqual } from 'react-redux';
import { useLocation } from 'react-router-dom';
import { useEffect } from 'react';
import Region from './Region/Region';
import { fetchRegions } from '../../Redux/regions/regions';
import loading from '../../Images/loading.gif';

const Regions = () => {
  const { pathname } = useLocation();

  const regionList = useSelector((state) => state.regionsReducer.regions, shallowEqual);
  const wait = useSelector((state) => state.regionsReducer.wait, shallowEqual);
  const dispatch = useDispatch();
  useEffect(() => {
    const countryName = pathname.replace('/', '');
    dispatch(fetchRegions(countryName));
  }, []);
  const date = new Date();
  const year = date.getFullYear();
  return (
    <>
      <h2
        className="white f6 tracked fw3 pa2"
        style={{
          background: '#d70060', textAlign: 'center', padding: '1rem', color: 'silver',
        }}
      >
        {`CITY/TOWN/REGION - ${year}`}
      </h2>
      {regionList.map((region) => (
        <Region
          key={region.id}
          name={region.name}
          newCases={region.newCases}
        />
      ))}
      {!regionList.length && !wait ? (
        <p
          className="white f4 fw4 tc pa3"
          style={{
            textAlign: 'center', color: 'silver', fontSize: '1.5rem', margin: '2rem, auto',
          }}
        >
          There is no regional data
        </p>
      ) : ''}
      {wait ? <img width="100" height="100" id="loading" className="db center" alt="loading" style={{ margin: '2rem 45%' }} src={loading} /> : ''}
    </>
  );
};

export default Regions;
