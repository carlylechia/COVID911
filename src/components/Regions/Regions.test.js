import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../Redux/createStore';
import Region from './Region/Region';
import Regions from './Regions';

describe('Regions component test', () => {
  test('test Regions component', () => {
    const props = {
      name: 'region 1',
      newCases: 50,
    };
    const {
      name, newCases,
    } = props;
    render(
      <Router>
        <Provider store={store}>
          <Region
            name={name}
            newCases={newCases}
          />
        </Provider>
      </Router>,
    );
    expect(render(
      <Router>
        <Provider store={store}>
          <Regions />
        </Provider>
      </Router>,
    )).toMatchSnapshot();
  });
});
