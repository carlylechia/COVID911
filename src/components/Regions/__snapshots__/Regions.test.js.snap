// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regions component test test Regions component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="f4 fw4 states-container pa5-ns pa3 pt3 pb3"
      >
        <p
          class="f4 fw4 white"
        >
          region 1
        </p>
        <p
          class="f4 fw4 tr pr3 white"
        >
          50
        </p>
      </div>
    </div>
    <div>
      <h2
        class="white f6 tracked fw3 pa2"
        style="background: rgb(215, 0, 96); text-align: center; padding: 1rem; color: silver;"
      >
        CITY/TOWN/REGION - 2022
      </h2>
      <img
        alt="loading"
        class="db center"
        height="100"
        id="loading"
        src="/images/loading.gif"
        style="margin: 2rem 45%;"
        width="100"
      />
    </div>
  </body>,
  "container": <div>
    <h2
      class="white f6 tracked fw3 pa2"
      style="background: rgb(215, 0, 96); text-align: center; padding: 1rem; color: silver;"
    >
      CITY/TOWN/REGION - 2022
    </h2>
    <img
      alt="loading"
      class="db center"
      height="100"
      id="loading"
      src="/images/loading.gif"
      style="margin: 2rem 45%;"
      width="100"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
